### PR TITLE
[VM] Minor refactor for C++ memory alloc

### PR DESCRIPTION
**Pull request by masahi**
Some leftover items from https://github.com/apache/tvm/pull/7369, to be cleaner and safer.

Instead of getting a pointer to packed func and call it indirectly, just declare C++ op function in a header and call it directly.

cc @zhiics @jroesch

### CODE REVIEW

File: src/relay/op/tensor/tensor.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+#ifndef TVM_RELAY_OP_TENSOR_TENSOR_H_
+#define TVM_RELAY_OP_TENSOR_TENSOR_H_
+
+#include "tvm/relay/expr.h"
+
+namespace tvm {
+namespace relay {
+
+Expr Prod(Expr data, Array<Integer> axis, bool keepdims, bool exclude);
```
Comment (zhiics) [inline src/relay/op/tensor/tensor.h:32]: BTW, we already have a bunch of op definitions in 'pattern_utils.h'. We can probably move this one there as well.

File: tests/python/frontend/pytorch/test_object_detection.py [hunk 2]
```diff
@@ -162,3 +162,8 @@ def compile_and_run_vm(mod, params, data_np, target):
     # Results should be equivalent after rewriting
     for res1, res2 in zip(tvm_res, tvm_res_after_rewrite):
         tvm.testing.assert_allclose(res1.asnumpy(), res2.asnumpy())
+
```
Comment (zhiics) [inline tests/python/frontend/pytorch/test_object_detection.py:165]: remove this and `in_size` above?

Comment (masahi) [reply tests/python/frontend/pytorch/test_object_detection.py:165]: oops will fix

File: src/relay/op/tensor/tensor.h (hunk 1 above)
Comment (masahi) [reply src/relay/op/tensor/tensor.h:32]: Thanks, didn't notice reduce ops are also defined there.

Comment (masahi) [reply src/relay/op/tensor/tensor.h:32]: But I do find that it is a bit weird to declare those make-op in `patterns_utils.`. `tensor.h` would be more appropriate, I think.

**Review by zhiics — APPROVED** [review]

**Pull request merged**